src/app/forms/validators/custom-async-validators.ts

```diff
@@ -114,7 +117,46 @@ export class CustomAsyncValidators {
 
           return null;
         })
+   
+        );
+  }
+
+  static hideIfEqualsWithCondition(store: Store<State>, sibling: string, value: string, conditions: Condition | Condition[]): AsyncValidatorFn {
+    return (control: AbstractControl): Observable<ValidationErrors | null> =>
+      store.pipe(
+        take(1),
+        select(testResultInEdit),
+        map(testResult => {
+          if (!testResult || !control?.parent) { return null; }
+
+          const siblingControl = control.parent.get(sibling) as CustomFormControl;
+
+          const conditionsPassed = CustomAsyncValidators.checkConditions(testResult, conditions);
+
+          siblingControl.meta.hide = conditionsPassed && (Array.isArray(value) ? value.includes(control.value) : control.value === value);
+
+          return null;
+        })
       );
   }
- 
+
+  static checkConditions(testResult: TestResultModel, conditions: Condition | Condition[] ){
+    if (!Array.isArray(conditions)) {
+      return CustomAsyncValidators.checkCondition(testResult, conditions);
+    }
+
+    return conditions.every(condition => CustomAsyncValidators.checkCondition(testResult,condition))
+  }
+
+  static checkCondition(testResult: TestResultModel, condition: Condition ){
+    const {field, operator, value} = condition
+
+    const fieldValue = testResult.testTypes[0].hasOwnProperty(field)
+      ? (testResult?.testTypes[0] as any)[field]
```

Comment:
```suggestion
      ? (testResult?.testTypes[0])[field as keyof TestType]
```

Reply:
I will leave this as is as it is difficult to get the change to work fully.